pkg/kubelet/cm/devicemanager/manager_test.go

```diff
@@ -351,18 +407,35 @@ func (a *activePodsStub) updateActivePods(newPods []*v1.Pod) {
 }
 
 type MockEndpoint struct {
-	allocateFunc func(devs []string) (*pluginapi.AllocateResponse, error)
+	allocateFunc            func(devs []string) (*pluginapi.AllocateResponse, error)
+	initChan                chan []string
+	getInitRequiredFlagFunc func() bool
 }
 
 func (m *MockEndpoint) stop() {}
 func (m *MockEndpoint) run()  {}
 
+func (m *MockEndpoint) getInitReqFlag() bool {
+	if m.getInitRequiredFlagFunc != nil {
+		return m.getInitRequiredFlagFunc()
+	}
+	return false
+}
+
 func (m *MockEndpoint) getDevices() []pluginapi.Device {
 	return []pluginapi.Device{}
 }
 
 func (m *MockEndpoint) callback(resourceName string, added, updated, deleted []pluginapi.Device) {}
 
+func (m *MockEndpoint) initialize(devs []string) (*pluginapi.Empty, error) {
+	if m.getInitRequiredFlagFunc != nil && m.getInitRequiredFlagFunc() {
```

Comment:
Shouldn't we actually test the endpoint? By making use of the device plugin stub?

Reply:
usin dp stub suits more in e2e tests. Since this is unit test, i think mock approach is appropriate.

Reply:
https://testing.googleblog.com/2015/04/just-say-no-to-more-end-to-end-tests.html